cartography/intel/kubernetes/ingresses.py

```diff
@@ -0,0 +1,108 @@
+from neo4j import Session
+
+from cartography.intel.kubernetes.util import get_epoch
+from cartography.intel.kubernetes.util import K8sClient
+from cartography.util import timeit
+
+logger = logging.getLogger(__name__)
+
+
+@timeit
+def sync_ingresses(
+    session: Session,
+    client: K8sClient,
+    update_tag: int,
+    cluster: Dict,
+) -> List[Dict]:
+    ingresses = get_ingresses(client)
+    ingresses_data = transform_ingresses(ingresses, cluster)
+    load_ingresses(session, ingresses_data, update_tag)
+    return ingresses
+
+
+@timeit
+def get_ingresses(client: K8sClient) -> V1IngressList:
+    return client.networking.list_ingress_for_all_namespaces().items
+
+
+@timeit
+def transform_ingresses(ingresses: V1IngressList, cluster: Dict) -> List[Dict]:
+    return [
+        {
+            "uid": ingress.metadata.uid,
+            "name": ingress.metadata.name,
+            "creation_timestamp": get_epoch(ingress.metadata.creation_timestamp),
+            "deletion_timestamp": get_epoch(ingress.metadata.deletion_timestamp),
+            "namespace": ingress.metadata.namespace,
+            "cluster_uid": cluster["uid"],
+            "labels": ingress.metadata.labels,
+            "rules": [
+                {
+                    "host": r.host,
+                    "http_paths": [
+                        {
+                            "path": p.path or "",
+                            "path_type": p.path_type if p.path else "Empty",
+                            "service": p.backend.service.name,
+                            "port": p.backend.service.port.number,
+                        }
+                        for p in r.http.paths
+                    ],
+                }
+                for r in ingress.spec.rules
+            ],
+            "tls": ingress.spec.tls,
+        }
+        for ingress in ingresses
+    ]
+
+
+def load_ingresses(session: Session, data: List[Dict], update_tag: int) -> None:
+    ingestion_cypher_query = """
+    UNWIND $ingresses as k8ingress
+        MERGE (ingress:KubernetesIngress {id: k8ingress.uid})
+        ON CREATE SET ingress.firstseen = timestamp()
+        SET ingress.lastupdated = $update_tag,
+            ingress.name = k8ingress.name,
+            ingress.created_at = k8ingress.creation_timestamp,
+            ingress.deleted_at = k8ingress.deletion_timestamp,
+            ingress.tls = k8ingress.tls
+        WITH ingress, k8ingress.namespace as ns, k8ingress.cluster_uid as cuid, k8ingress.rules as k8rules
+        MATCH (cluster:KubernetesCluster {id: cuid})-[:HAS_NAMESPACE]->(space:KubernetesNamespace {name: ns})
+        MERGE (space)-[rel1:HAS_INGRESS]->(ingress)
+        ON CREATE SET rel1.firstseen = timestamp()
+        SET rel1.lastupdated = $update_tag
+        WITH space, ingress, k8rules
+        UNWIND k8rules as k8rule
+            MERGE (rule: KubernetesIngressRule {ingress_uid: ingress.id, host: k8rule.host})
```

Comment:
Just like you did for `KubernetesIngress`, I still need you to create some suitable `id` for for `KubernetesIngressRule` and for `KubernetesIngressRuleHttpPath`.

Reply:
I still cannot see how an `id` would be needed when `id`s, for rules and paths, are in fact composed indexes, but I won't have time to continue this discussion further anytime soon. Thanks for the review anyway.